docs/src/ToricVarieties/cohomCalg.md

```diff
@@ -0,0 +1,77 @@
+```@meta
+CurrentModule = Oscar
+```
+
+```@contents
+Pages = ["CohomCalg.md"]
+```
+
+
+# Line bundle cohomology with cohomCalg
+
+We employ the cohomCalg algorithm [cohomCalg:Implementation](@cite)
+to compute the dimension of line bundle cohomologies as well as vanishing sets.
+
+
+## Dimensions of line bundle cohomology
+
+```@docs
+all_cohomologies(l::ToricLineBundle)
+cohomology(l::ToricLineBundle, i::Int)
+```
+
+## Vanishing sets
+
+### Shifted cones
+
+```@docs
+ShiftedCone(gens::Vector{Vector{Int64}}, shift::Vector{Int64})
+```
+
+Note that a shifted cone is trivial iff there are no generators for
+the cone `C` and there is no affine offset. We support the method
+`istrivial(C::ShiftedCone)` to check this.
+
+Equality of shifted cones `sc1` and `sc2` is supported via `sc1 == sc2`.
+
+We support the following attributes for shifted cones:
+```@docs
+gens(sc::ShiftedCone)
+ngens(sc::ShiftedCone)
+ambient_dim(sc::ShiftedCone)
+offset(sc::ShiftedCone)
+cone(sc::ShiftedCone)
+```
+
+### Toric vanishing sets
+
+Vanishing sets of toric varieties based on [cohomCalg:Implementation](@cite)
+were originally introduced in [Bie18](@cite). They are the complement of
+a finite family of shifted cones, as introduced in the previous section.
+
+We construct such a vanishing set by issuing
+`ToricVanishingSet(v::AbstractNormalToricVariety, scs::Vector{ShiftedCone}, i::fmpz)`.
+This vanishing set then describes all line bundle bundles on the toric variety `v` for which
+the `i`-th sheaf cohomology group vanishes.
+
+Given a toric vanishing set `tvs`,  `isfull(tvs)` returns `true` if the vanishing set covers
+the entire Picard group of its toric variety. Otherwise, it returns `false`.
+
+The following attributes are supported for toric vanishing sets:
+```@docs
+toric_variety(tvs::ToricVanishingSet)
+shifted_cones(tvs::ToricVanishingSet)
+cohomology_index(tvs::ToricVanishingSet)
+```
+
+Of ample importance is the ability to check if a toric line bundle is contained in a
+toric vanishing set:
+```@docs
+contains(tvs::ToricVanishingSet, l::ToricLineBundle)
```

Comment:
Sorry for the stupid question (I have *no* idea what these objects are), but to double check: so this means `l` is an *element` of `tvs`, not a subset, yes?

Reply:
The short answer is "yes".

`l` represents a line bundle. Collectively, all line bundles on a toric variety (or any geometric space) form the Picard group. The vanishing set describes a subsets of the Picard group. The question for `contains` is to tell if `l` is an element of the vanishing set or not.

(I suppose, one could consider `l` as a one-element subset of `tvs`. But this seems besides the point I guess. In any case, `l` is at most one element of the vanishing set `tvs`.)

I suppose I should state this more clearly in the documentation.

Reply:
I have just done exactly this with a rebase.